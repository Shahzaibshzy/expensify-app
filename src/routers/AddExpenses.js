import React from "react";
import { connect } from "react-redux";
import { addExpense } from "../actions/expenses";
import ExpenseForm from "./ExpenseForm";
import { useNavigate } from "react-router-dom";

const AddExpensesFunctional = ({ onSubmit }) => {
  const navigate = useNavigate();
  const handleSubmit = (expense) => {
    onSubmit(expense);
    navigate("/");
  };

  return (
    <div>
      this is my expenses addition
      <h1>Add Expense</h1>
      <ExpenseForm onSubmit={handleSubmit} />
    </div>
  );
};

export class AddExpenses extends React.Component {
  render() {
    return (
      <div>
        <AddExpensesFunctional onSubmit={this.props.addExpense} />
      </div>
    );
  }
}
const mapDispatchToProps = (dispatch) => ({
  addExpense: (expense) => dispatch(addExpense(expense)),
});
export default connect(undefined, mapDispatchToProps)(AddExpenses);
