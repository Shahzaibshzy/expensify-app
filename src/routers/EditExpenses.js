import React from "react";
import { useParams } from "react-router-dom";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { editExpense } from "../actions/expenses";
import { useNavigate } from "react-router-dom";
import { remoeExpense } from "../actions/expenses";

const EditExpenses = (props) => {
  const { id } = useParams();
  const navigate = useNavigate();

  // Example usage of navigate to go back to the homepage ('/')
  const handleGoBack = () => {
    navigate("/");
  };


  return(<div>
    <ExpenseForm expense={props.expense} onSubmit={(expense) => {
      props.dispatch(editExpense(id,expense));
      handleGoBack();
    }} />
    <button
      onClick={() => {
        props.dispatch(remoeExpense({ id }));
        handleGoBack();
      }}
    >
      Remove
    </button>
    
    </div>)
};

const mapStateToProps = (state) => {
  const params = {id: window.location.pathname.split("/")[2]}
  return {
      expense: state.expenses.find((expense) => expense.id === params.id)
  }
}

export default connect(mapStateToProps)(EditExpenses);
