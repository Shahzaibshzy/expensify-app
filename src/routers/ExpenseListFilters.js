import React from "react";
import { connect } from "react-redux";
import { setTextFilter, sortByDate, sortByAmount , setStartDate , setEndDate} from "../actions/filters";
import { DateRangePicker } from "react-dates";

class ExpenseListFilters extends  React.Component {
  state = {

    calendarFocused: null,

  }
 onDateChange = ({startDate, endDate}) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  };
  onFocusChange = (calendarFocused) => {
    this.setState(()=>({calendarFocused}))
  };
  render(){

    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={(e) => {
            this.props.dispatch(setTextFilter(e.target.value));
          }}
        />
        <select
          value={this.props.filters.sortBy}
          onChange={(e) => {
            if (e.target.value === "date") {
              this.props.dispatch(sortByDate());
            } else if (e.target.value === "amount") {
              this.props.dispatch(sortByAmount());
            } // sort by date first if amount is
          }}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker 
        startDate={this.props.filters.startDate}
        startDateId="MyDatePickerStart" // PropTypes.string.isRequired,
        endDate={this.props.filters.endDate}
        endDateId= "MyDatePickerEnd" // PropTypes.string.isRequired,
        onDatesChange={this.onDateChange}
        focusedInput={this.state.calendarFocused}
        onFocusChange={this.onFocusChange}
        showClearDates={true}
        numberOfMonths={1}
        isOutsideRange={()=> false}

        
        />
      
      </div>
    );

  }


}

const mapStateToProp = (state) => {
  return {
    filters: state.filters,
  };
};

export default connect(mapStateToProp)(ExpenseListFilters);
