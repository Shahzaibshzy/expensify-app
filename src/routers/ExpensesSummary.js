import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import selectExpenses from "../selectors/expenses";
import totalexpenses from "../selectors/expenses-total";


export const ExpensesSummary = ({ expenseCount, expenseTotal }) => {
  const expenseWord = expenseCount === 1 ? "expense" : "expenses";
  const expenseTotalFormat = numeral(expenseTotal / 100).format("$0,0.00");

  return (
    <div>
      <h1>
        viewing {expenseCount} {expenseWord} totalling {expenseTotalFormat}
      </h1>
    </div>
  );
};

const mapStateToProp = (state) => {
const VisibleExpense = selectExpenses(state.expenses, state.filters);
return {
    expenseCount: VisibleExpense.length,
    expenseTotal: totalexpenses(VisibleExpense)
}
};

export default connect(mapStateToProp)(ExpensesSummary);