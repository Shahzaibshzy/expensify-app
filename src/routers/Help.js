import React from "react";
const Help = () => <div>this is my expenses help</div>;

export default Help;
