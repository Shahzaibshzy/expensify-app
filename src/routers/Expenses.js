import React from "react";
import ExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";

const Expenses = () => (
  <div>
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default Expenses;
