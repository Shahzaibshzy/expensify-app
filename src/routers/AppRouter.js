import AddExpenses from "./AddExpenses";
import EditExpenses from "./EditExpenses";
import Expenses from "./Expenses";
import Header from "./Header";
import Help from "./Help";
import NotFoundPage from "./NotFoundPage";
import React from "react";
import { BrowserRouter, Route, Routes } from "react-router-dom";

const AppRouter = () => (
  <BrowserRouter>
    <div>
      <Header />
      <Routes>
        <Route path="/" element={<Expenses />} />
        <Route path="/create" element={<AddExpenses />} />
        <Route path="/edit/:id" element={<EditExpenses />} />
        <Route path="/help" element={<Help />} />
        <Route path="*" element={<NotFoundPage />} />
      </Routes>
    </div>
  </BrowserRouter>
);

export default AppRouter;
