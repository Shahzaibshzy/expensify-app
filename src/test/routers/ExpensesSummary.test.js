import React from "react";
import { shallow } from "enzyme";
import {ExpensesSummary} from "../../routers/ExpensesSummary";

test("should have render ExpensesSummary with 1 expense", () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1}  expenseTotal = {295}/>);
  expect(wrapper).toMatchSnapshot();
});

test("should have render ExpensesSummary with multiple expenses", () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={23} expenseTotal={12312323} />);
  expect(wrapper).toMatchSnapshot();    
});
