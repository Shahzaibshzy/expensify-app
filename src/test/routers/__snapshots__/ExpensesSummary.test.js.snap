// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    viewing 
    1
     
    expense
     totalling 
    $2.95
  </h1>
</div>
`;

exports[`should have render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    viewing 
    23
     
    expenses
     totalling 
    $123,123.23
  </h1>
</div>
`;
